Log when a course is updated

renatho opened the pull request:
Fixes #2796

### Changes proposed in this Pull Request

* Add log for course update.
  * It ignores REST requests, similar to the `sensei_course_publish ` event.
  * In a conversation with @yscik, he suggested the idea to skip the post when it's being saved with a resave needed (when saves course outline block, needing to save again for ids sync). So before to log, we check if all lessons and modules contain ID (if not, a new post is coming).
  * The number of the lessons is counting only the published lessons, similar to the current `sensei_course_publish `. Is that the idea?
    * If we think this information is important for this log, maybe we could log the published lessons and the draft lessons separately.

### Testing instructions

See https://github.com/Automattic/sensei/issues/2796 to understand the correct values to be logged.

* Create a course without course outline block.
* Save and make sure it's logged with the correct values once.
* Change some information.
* Save again and make sure it's logged with the correct values one more time.
* Add the course outline block.
* Save again and make sure it's logged with the correct values one more time.
* Add / change / remove modules and lessons.
* Save again and make sure it's logged with the correct values one more time.
* Activate the WCPC, create some products, add/remove them to the product.
* Save again and make sure it's logged with the correct values one more time.

renatho commented on includes/class-sensei-course.php:
```diff
@@ -3459,6 +3475,103 @@ public function log_initial_publish_event( $course ) {
+			'has_outline_block' => has_block( 'sensei-lms/course-outline', $content ) ? 1 : 0,
+			'module_count'      => count( wp_get_post_terms( $course_id, 'module' ) ),
+			'lesson_count'      => $this->course_lesson_count( $course_id ),
+			'product_count'     => $product_count,
```
We could create a filter for that, and add this feature through the WCPC, but as it only for log I think it wouldn't make sense. And it follows the same idea used in the `sensei_course_publish `

donnapep commented:
> * The number of the lessons is counting only the published lessons, similar to the current `sensei_course_publish `. Is that the idea?

Yes, that's correct.

The first test I tried was to go to an existing course that is not using the outline block, and clicked _Update_. This did not log the `sensei_course_update` event, but it did log `sensei_course_publish`, which is backwards. `sensei_course_update` should have been logged but not `sensei_course_publish`.

renatho commented:
@donnapep Ah, my bad! I had added the event with the wrong name. It's fixed here: https://github.com/Automattic/sensei/pull/3652/commits/b4cc71aeea907f73a4a8cb547f811cfb89b5d3cd

donnapep commented:
Working now, but found a new problem. 😄  This event should only fire if the _Publish_ or _Update_ button is clicked; in other words, when the course is live on the site and not "under construction". When I clicked on _Save draft_ the event was logged. Same thing with _Save as pending_.

renatho commented:
> Working now, but found a new problem. 😄 This event should only fire if the _Publish_ or _Update_ button is clicked; in other words, when the course is live on the site and not "under construction". When I clicked on _Save draft_ the event was logged. Same thing with _Save as pending_.

Hum! Perfect! I updated to log only when the course status is `publish` and I renamed some things to make more sense. https://github.com/Automattic/sensei/pull/3652/commits/7ea32375398000a14aa6a99759a978d53338251a

donnapep reviewed (approved):
Hope you don't mind me adding a commit to your PR. 😃 I've tested everything and it works great! 🎉
